core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -1029,21 +1030,54 @@ static void loadDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePa
     if (inodePath.fullPathExists()) {
       return;
     }
+    MountTable.Resolution resolution = mountTable.resolve(inodePath.getUri());
+    // create the actual metadata
+    loadDirectoryMetadataInternal(rpcContext, context, inodePath, resolution.getUri(),
+        resolution.getUfsClient(), fsMaster, resolution.getShared(),
+        mountTable.isMountPoint(inodePath.getUri()));
+  }
+
+  /**
+   * Loads metadata for the directory that will be the mount point.
+   *
+   * This difference between this method and
+   * {@link InodeSyncStream#loadDirectoryMetadata(RpcContext, LockedInodePath, LoadMetadataContext,
+   * MountTable, DefaultFileSystemMaster)} is that it accepts more specific parameters(i.e
+   * . isShared, ufsUri and ufsClient) and doesn't require the MountTable as its parameter.
+   */
+  static void loadMountPointDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePath,
+      LoadMetadataContext context, boolean isShared, AlluxioURI ufsUri,
+      UfsManager.UfsClient ufsClient, DefaultFileSystemMaster fsMaster) throws
+      FileDoesNotExistException, IOException, InvalidPathException, AccessControlException {
+    if (inodePath.fullPathExists()) {
+      return;
+    }
+    // create the actual metadata
+    loadDirectoryMetadataInternal(rpcContext, context, inodePath, ufsUri,
+        ufsClient, fsMaster, true, isShared);
```

Comment:
`true, isShared` <- is this argument order correct? Should be `isShared, true`?

Reply:
This one is correct. But L1032 is wrong. I will fix it.